illumina2vcf/vcf.py

```diff
@@ -18,13 +18,85 @@ class ConverterError(Exception):
     pass
 
 
+class QC_stats:
+    def __init__(self):
+        self.vcf_lines_autosomal = 0
+        self.called_lines_autosomal = 0
+        self.heterozygous_lines_autosomal = 0
+        self.called_lines_x = 0
+        self.heterozygous_lines_x = 0
+        self.vcf_lines_y = 0
+        self.called_lines_y = 0
+
+    def vcf_comment(self) -> str:
+        qc_stats = {}
+        try:
+            qc_stats["callrate"] = self.call_rate("autosomal")
+        except ZeroDivisionError:
+            pass
+        try:
+            qc_stats["y_notnull"] = self.call_rate("Y")
+        except ZeroDivisionError:
+            pass
+        try:
+            qc_stats["het"] = self.heterozygosity("autosomal")
+        except ZeroDivisionError:
+            pass
+        try:
+            qc_stats["x_het"] = self.heterozygosity("X")
+        except ZeroDivisionError:
+            pass
+        if qc_stats:
+            return VCFLine.as_comment_key_dict("qc_stats", qc_stats)
+        else:
+            pass
+
+    def call_rate(self, chrom: Literal["autosomal", "Y"]) -> float:
+        if chrom == "autosomal":
+            return "{0:.2f}".format(self.called_lines_autosomal / self.vcf_lines_autosomal)
```

Comment:
🔧 This can be done with f-strings too: `f"{self.called_lines_autosomal / self.vcf_lines_autosomal:.2f}"`.

Also, this will return a string, but the type hint says its returning a float. Not sure if its cleaner to do the string conversion here, or in `vcf_comment()` but it should be consistently annotated one way or the other.